index.html

```diff
@@ -296,141 +292,170 @@
+    switch (days[0][day][school][currentSlot]["period"]) { // Falling through intended
+      case "0": // 0 Period
+      case "A": // A period
+      case "B": // B period
+      case "C": // C period
+      case "D": // D period
+      case "E": // E period
+      case "F": // F period
+      case "G": // G period
+      case "H": // H period
+        for (var k = 0; k < schedule["classes"].length; k++) {
+          if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
+            var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
+            var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
+            scheduleObj = { name: schedule["classes"][k]["name"],
+              teacher: teacherName,
+              teacherLastName: teacherLast,
+              room: schedule["classes"][k]["room"],
+              period: days[0][day][school][currentSlot]["period"],
+              time: ""};
+            break;
+          }
         }
-        // Add in photos if the user's schedule is being rendered
-        if (type == "full") {
+        break;
 
-          // If the user is a normal user and the class is a normal class
-          if (scheduleObj.teacher != "" && schedule.grade) {
-            var sanitized_teacher_name;
-            sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
-            sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-            scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
+      default: // If it is a special class, such as Lunch, Assembly, or Class Meeting
+      // Note that 0 Period Instrumental Music does not fall in this category
+        scheduleObj = {
+          name: days[0][day][school][currentSlot]["period"],
+          teacher: "",
+          room: "",
+          period: "X",
+          time: ""
+        };
+        if (scheduleObj.name == "Lunch" ||
+          scheduleObj.name == "Assembly" ||
+          scheduleObj.name == "US Community") {
 
-          // If class is a special class or free period
-          } else if (scheduleObj.period == "X" || scheduleObj.name == "Free Period") {
-            scheduleObj.teacherLink = "";
-            var image_url = scheduleObj.name.toLowerCase();
-            image_url = image_url.replace("/", "");
-            image_url = image_url.replace(/\s/g, '');
-            scheduleObj.avatar = "/images/" + image_url + ".jpg";
+          scheduleObj.room = "LPC Commons";
+        }
+    }
+    if (scheduleObj.teacher == "N/A") {
+      scheduleObj.teacher = "";
+    }
+    if (scheduleObj.room == "N/A") {
+      scheduleObj.room = "";
+    }
+    // Add in photos if the user's schedule is being rendered
+    if (type == "full") {
+      // Add the left-hand images to the schedule
+      scheduleObj = addScheduleImages(scheduleObj, schedule);
```

Comment:
Don't need to use a return value from addScheduleImages. That function will change scheduleObj in-place.